charts/vector/templates/haproxy/_helpers.tpl

```diff
@@ -38,3 +38,14 @@ Create the name of the service account to use
 {{- default "default" .Values.haproxy.serviceAccount.name }}
 {{- end }}
 {{- end }}
+
+{{/*
+Return the appropriate apiVersion for HPA autoscaling APIs.
+*/}}
+{{- define "autoscaling.apiVersion" -}}
+{{- if or (.Capabilities.APIVersions.Has "autoscaling/v2") (semverCompare ">=1.23" .Capabilities.KubeVersion.Version) -}}
+"autoscaling/v2"
+{{- else -}}
+"autoscaling/v2beta2"
```

Comment:
I think we'll want to keep a check for `v2beta2` support to not cause issues with clusters that are so old that they don't even support that version. Though I'm not seeing how far back that would end up being.

Reply:
v2beta2 is available from k8s 1.12, and it'll require some changes in HPA config (targetAverageUtilization vs averageUtilization) if we go back to v2beta1, while vector only support k8s 1.15+. I don't see why we really need to go back for v2beta1 support for now.

Reply:
Thanks - that's definitely far back enough!